src/app/news.service.ts

```diff
@@ -1,41 +1,42 @@
 // tslint:disable: variable-name
 import { Injectable } from '@angular/core';
 import { combineLatest, OperatorFunction, Subject } from 'rxjs';
-import { map, scan, shareReplay, startWith } from 'rxjs/operators';
+import {map, scan, shareReplay, startWith} from 'rxjs/operators';
 
 @Injectable({
   providedIn: 'root'
 })
 export class NewsService {
-  private _news$ = new Subject<string>();
-  private _breakingNews$ = new Subject<string>();
+  private _news$ = new Subject<{title: string, date: number}>();
+  private _breakingNews$ = new Subject<{title: string, date: number}>();
 
-  news$ = this._news$.pipe(storeNews());
-  breakingNews$ = this._breakingNews$.pipe(storeNews());
+  news$ = this._news$.pipe(storeNews(), shareReplay(1));
+  breakingNews$ = this._breakingNews$.pipe(storeNews(), shareReplay(1));
 
   allNews$ = combineLatest(
-    // combineLatest will start emitting when all of it's streams will emit at least one time
-    // that is why startWith() operator is required here - we want to emit normal news
-    // even when there were no breaking news yet and vice versa
     this.breakingNews$.pipe(startWith([])),
     this.news$.pipe(startWith([]))
-  ).pipe(map(([breaking, news]) => [...breaking, ...news]));
+  ).pipe(
+    map(([breaking, news]) => {
+      return [...breaking, ...news].sort((a, b) => a.date - b.date);
+    }),
```

Comment:
This is interesting approach with capturing dates and sorting.
However there is another way without sorting by just naturally arranging streams in a way that does not require separation - by using `merge` operator on both news streams and then applying `storeNews` operator like so:
```ts
allNews$ = merge(this.breakingNews$, this.news$).pipe(storeNews())
```

Reply:
ok, thanks )